spark/spark-2.1/src/main/scala/org/apache/spark/sql/hive/HivemallOps.scala

```diff
@@ -805,6 +805,47 @@ final class HivemallOps(df: DataFrame) extends Logging {
     JoinTopK(kInt, df.logicalPlan, right.logicalPlan, Inner, Option(joinExprs.expr))(score.named)
   }
 
+  private def doFlatten(schema: StructType, prefix: Option[String] = None) : Seq[Column] = {
+    schema.fields.flatMap { f =>
+      val colName = prefix.map(p => s"$p.${f.name}").getOrElse(f.name)
```

Comment:
Dot `.` is a special symbol in SQL. So, better to change the default separator (e.g., `$` ) and have an option to choose the separator. `news20.train` means train table in news20 database.

SQL identifiers are depending on RDBMSs but generally includes `[0-9,a-z,A-Z$_]`.
https://www.postgresql.org/docs/9.1/static/sql-syntax-lexical.html
https://msdn.microsoft.com/en-us/library/ms175874.aspx
https://dev.mysql.com/doc/refman/5.7/en/identifiers.html